### DISCUSSION TITLE
concatenate strings give errors (ex. select fname || ' ' || lname)

### DISCUSSION THREAD

**Discussion by fizzz71** [dbeaver/dbeaver#22978] [Q&A] [answered]
### Description

select fname || ' ' || lname from CLIENT 
(gives : SQL Error [335544849] [22000]: Malformed string [SQLState:22000, ISC error code:335544849])

select fname || lname from CLIENT (is working)

### DBeaver Version

23.3.5 & 23.2 (and all version before)

### Operating System

windows 10

### Database and driver

firebird 2.5 

### Steps to reproduce

select fname || ' ' || lname from CLIENT 
(gives : SQL Error [335544849] [22000]: Malformed string [SQLState:22000, ISC error code:335544849])

select fname || lname from CLIENT (is working)

**Comment by E1izabeth** [comment]
I can't reproduce the issue.
What are your Server and Driver version?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/52a73479-9016-4bc3-ba2b-2ced235e5494)

**Comment by fizzz71** [comment]
We use Windows ODBC 
And in ODBC, it is Firebird/InterBase(r) driver version 2.00.05.156

here is my test connection screenshot:
<img width="365" alt="Screenshot 2024-02-26 at 15 41 51" src="https://github.com/dbeaver/dbeaver/assets/17870502/11af6b17-9f44-4362-bae8-abbb6f2bb7ff">

**Comment by E1izabeth** [comment]
Hmm.. I tried decreasing the driver version to 4.0.0, but still can't reproduce. So, at least, I can say that it doesn't appear to be a problem with the driver.

**Comment by fizzz71** [comment]
That's weird. On the same machine and same select statement, I tested it out in SQL Workbench, with the same jaybird driver version, and it worked. 

And it also worked in isql.exe

**Comment by fizzz71** [comment, marked as answer]
SOLVED:  i needed to set encoding to ISO8859_1 (in the connection configuration, and instead of UTF8)
